File: dict.m
t = 0:0.01:1;

sin_dict = struct('s1', [2, 5], 's2', [11, 17], 's3', [12, 20], 's4', [20, 30], 's5', [24, 50]);

h = input('Enter sinusoidal key to generate: ', 's');

if isfield(sin_dict, h)
    x = sin_dict.(h)(1) * sin(2 * pi * sin_dict.(h)(2) * t);
    plot(t, x);
    xlabel('Time');
    ylabel('Amplitude');
    title(['Sinusoidal Signal for ', h]);
    grid on;
    legend(h);
    axis tight;
    show();
else
    disp('Invalid key entered. Please choose from the following keys:');
    disp(fieldnames(sin_dict)');
end

File: dict.py
import numpy as np
from matplotlib import pyplot as plt
t=np.arange(0,1,0.01)
plt.xlabel('time')
plt.ylabel('amplitute')
plt.title('sine wave')
k=input("enter your wave:")
dic={'s1':[2,5],'s2':[5,10],'s3':[3,7],'s4':[10,12],'s5':[1,2]}
if dic[k]:
	x=dic[k][0]*np.sin(2*np.pi*dic[k][1]*t)
	plt.plot(t,x)
	plt.show()

